html/janus.js

```diff
@@ -1864,6 +1864,29 @@ function Janus(gatewayCallbacks) {
 		callbacks = callbacks || {};
 		callbacks.success = (typeof callbacks.success == "function") ? callbacks.success : Janus.noop;
 		callbacks.error = (typeof callbacks.error == "function") ? callbacks.error : webrtcError;
+
+		function stopTracksOnError(stream, callbacks) {
+			var originalErrorCallback = callbacks.error;
+			callbacks.error = function (err) {
+				try {
+					// Try a MediaStreamTrack.stop() for each track
+					var tracks = stream.getTracks();
+					for(var mst of tracks) {
+						Janus.log(mst);
+						if(mst)
+							mst.stop();
+					}
+				} catch(e) {
+					// Do nothing if this fails
+				}
+				if (typeof callbacks.error == "function") {
```

Comment:
at this point `callbacks.error` is always a function because of `callbacks.error = (typeof callbacks.error == "function") ? callbacks.error : webrtcError;` above. This is just a additional safety check.